Common/Tools/trackTuner.h

```diff
@@ -0,0 +1,679 @@
+
+  std::unique_ptr<TGraphErrors> grOneOverPtPionCurrent;
+  std::unique_ptr<TGraphErrors> grOneOverPtPionUpgr;
+
+  std::unique_ptr<TGraphErrors> grDcaXYPullVsPtPionCurrent;
+  std::unique_ptr<TGraphErrors> grDcaXYPullVsPtPionUpgr;
+
+  std::unique_ptr<TGraphErrors> grDcaZPullVsPtPionCurrent;
+  std::unique_ptr<TGraphErrors> grDcaZPullVsPtPionUpgr;
+
+  /// @brief Function to configure the TrackTuner parameters
+  /// @param inputString Input string with all parameter configuration. Format: <name>=<value>|<name>=<value>
+  /// @return String with the values of all parameters after configurations are listed, to cross check that everything worked well
+  std::string configParams(std::string inputString)
+  {
+
+    std::string delimiter = "|";
+    std::string assignmentSymbol = "=";
+
+    LOG(info) << "[TrackTuner] === ";
+    LOG(info) << "[TrackTuner] === Parameter configuration via std::string";
+    LOG(info) << "[TrackTuner] === Required format: \"<name>" << assignmentSymbol << "<value>" << delimiter << "<name>" << assignmentSymbol << "<value>" << delimiter << "<name>" << assignmentSymbol << "<value>\"";
+    LOG(info) << "[TrackTuner] === Delimiter symbol: \"" << delimiter << "\"";
+    LOG(info) << "[TrackTuner] === Assignment symbol: \"" << assignmentSymbol << "\"";
+    LOG(info) << "[TrackTuner] === ";
+    LOG(info) << "[TrackTuner]";
+    LOG(info) << "[TrackTuner] >>> Original input string = \"" << inputString << "\"";
+
+    /// Check the format of the input string
+    if (inputString.find(delimiter) == std::string::npos) {
+      // wrong delimiter symbol used
+      LOG(fatal) << "ERROR: delimiter symbol \"" << delimiter << "\" not found in the configuration string. Fix it!";
+    }
+    if (inputString.find(assignmentSymbol) == std::string::npos) {
+      // wrong assignment symbol used
+      LOG(fatal) << "ERROR: assignment symbol \"" << assignmentSymbol << "\" not found in the configuration string. Fix it!";
+    }
+    int spaces = std::count(inputString.begin(), inputString.end(), ' ');
+    if (spaces > 0) {
+      // white spaces to be removed
+      LOG(fatal) << "ERROR: " << spaces << " white spaces found in the configuration string. Remove them!";
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////
+    /// Parameters to be configured via the string
+    /// +++ to be manually updated every time one adds a new parameter to the TrackTuner.h +++
+    enum ePars { eDebugInfo = 0,
+                 eUpdateTrackCovMat,
+                 eUpdateCurvature,
+                 eUpdatePulls,
+                 ePathInputFile,
+                 eIsInputFileFromCCDB,
+                 eNameInputFile,
+                 eUsePvRefitCorrections,
+                 eOneOverPtCurrent,
+                 eOneOverPtUpgr,
+                 eNPars };
+    std::map<uint8_t, std::string> mapParNames = {
+      std::make_pair(static_cast<uint8_t>(eDebugInfo), "debugInfo"),
```

Comment:
```suggestion
      std::make_pair(eDebugInfo, "debugInfo"),
```
The cast should not be needed, if the enum values are declared as `uint8_t`.